File: codemap/indexer/testdata/scala/src/main/scala/acme/orders/OrderService.scala
package acme.orders

import acme.store.{PgRepo, Repo}
import acme.util._

/** Places orders. */
class OrderService(repo: Repo) {
  def place(order: Order): Int = {
    if (!validate(order)) return 0
    val pg = new PgRepo()
    pg.insert(order)
    clean(order)
    repo.save(order) + PgRepo.table.length
  }

  private def validate(order: Order): Boolean = order.id > 0
}

object OrderService {
  def apply(): OrderService = new OrderService(new PgRepo())
}

case class Order(id: Int)

object Main {
  def run(): Int = OrderService().place(Order(1))
}

File: codemap/indexer/testdata/kt/app/src/main/kotlin/acme/orders/OrderService.kt
package acme.orders

import acme.store.PgRepo
import acme.store.Repo
import acme.util.*

/** Places orders. */
class OrderService(private val repo: Repo) {
    constructor() : this(PgRepo())

    fun place(o: Order): Int {
        if (!validate(o)) {
            return 0
        }
        val pg = PgRepo()
        pg.insert(o)
        clean(o.id)
        return repo.save(o) + PgRepo.table().length
    }

    private fun validate(o: Order) = o.id > 0
}

data class Order(val id: Int)

fun run() = OrderService().place(Order(1))
